### ISSUE TITLE
Pyright can't infer the type of a property clone

### ISSUE THREAD

**Issue by gpanago** [microsoft/pyright#7604]
It is seems that pyright can't infer the type from the following code and does not complain that int has no member "random":

```
def property_clone(fget, fset=None, fdel=None, doc=None):
    return property(fget=fget, fset=fset, fdel=fdel, doc=doc)
class C:
    @property_clone
    def a(self):
        return 3

C().a.random()
```

Pyright behaves correctly if I instead use the standard property:

```
class C:
    @property
    def a(self):
        return 3

C().a.random()
```

**Comment by erictraut** [comment]
Pyright is working as intended here, so I don't consider this a bug.

Static type checkers like pyright and mypy require special-casing for the `property` class. The special-case logic is extensive, and it applies only to `property` and only when `@property` is used as a direct decorator. 

The reason that so much special-case logic is required is that the `property` class should have probably been defined as a generic class, but it wasn't. It also has a bunch of internal state that can't generally be represented in the type system.

If you want your code to work with a static type checker, you should either stick to the standard usage of the built-in `property` or define your own descriptor class. Wrapping the built-in `property` will not work, as you've discovered.
